### PR TITLE
Netstandard Support

### LINKED ISSUE #9: NetStandard Support
This library should be updated to support netstandard. This would allow it to be using in newer netcoreapp unit tests, as well as Xamarin targets.

### PR DESCRIPTION
**Pull request by dansiegel**
# Description

Updates project to multi-target keeping legacy net45 support, while adding targets for netstandard2.0 and uap10.0.16299. This ensures that unit tests using the full framework, netcoreapp2.0+ or UWP will each have a supporting binary.

fixes #9 

## Packaging

The project structure has been updated to support the newer Sdk Style builds. By doing this, there is no longer a need for the nuspec, and a NuGet package is generated on each build. In addition to generating a singular NuGet for distribution, I have enabled generating a Symbols package and SourceLink each of which will better support debugging by allowing package consumers to step into the code for the MockServer. By default all generated packages will be generated in an Artifacts folder in the solution root, however if built on VSTS it will default to the Artifact Staging Directory.

## Versioning

Using the newer Sdk Style project format also allows each build to better version the builds for CI. The root version is controlled in the Directory.build.props, beyond that the assembly and package versions are controlled by environment variables. When built on either AppVeyor or VSTS this will result in a format like 1.5.0.{BuildNumber}. Each build by default will add the suffix `ci`. If an environment variable is present on the build server or passed in as a command line argument this will enable building a preview or release package.

- Default (generates a package like 1.5.0.{BuildNumber}-ci)
- IS_PREVIEW : true (generates a package like 1.5.0.{BuildNumber}-pre)
- IS_RELEASE : true (generates a package like 1.5.0.{BuildNumber})

From the command line it can be added like:

```
msbuild /p:Configuration=Release /p:IS_PREVIEW=true
```

## Signing

Since this makes major changes to the Project files, I have pulled in the snk from PR #8 and enabled assembly signing to support use in strongly signed assemblies. 

## CI

I've updated the AppVeyor YAML, but for some reason I was still getting a build error. I have also included a VSTS YAML file that I can confirm as working.

### CODE REVIEW

File: MockHttpServer.UnitTests/MockServerTests.cs [hunk 1]
```diff
@@ -306,18 +306,19 @@ public void TestSpecifiedMethods()
...
-            var client = CreateRestClient();
-            string expectedResult = "Result";
-
-            using (new MockServer(TestPort, "", (req, rsp, prm) => expectedResult, "+"))
-            {
-                var result = client.Execute(new RestRequest("", Method.POST));
-                Assert.AreEqual(expectedResult, result.Content);
-            }
-        }
+        // TODO: This test didn't work to begin with
```
Comment (jrharmon) [inline MockHttpServer.UnitTests/MockServerTests.cs]: Any details on what the error you were seeing was?  The test was definitely passing before, as you can see from the most recent Appveyor logs.

**Comment by jrharmon** [comment]
Thanks for the great work on this PR!  I added one comment about the unit test you commented out, but other than that everything looks great.  Once we settle on what to do with that, I'll merge it in.

**Comment by alexdresko** [comment]
Amazing work, y'all.

File: MockHttpServer.UnitTests/MockServerTests.cs (hunk 1 above)
Comment (dansiegel) [reply MockHttpServer.UnitTests/MockServerTests.cs]: Interestingly when I hooked up AppVeyor the test on master passes, but when I ran locally it fails. The error was as follows:

```
MockHttpServer.UnitTests.MockServerTests.TestSpecifiedHostName threw exception: 
System.Net.HttpListenerException: Access is denied
```

Comment (alexdresko) [reply MockHttpServer.UnitTests/MockServerTests.cs]: @dansiegel Try running VS as an administrator.

Comment (dansiegel) [reply MockHttpServer.UnitTests/MockServerTests.cs]: well crap that actually does work... I'll back that change out.

**Comment by dansiegel** [comment]
@jrharmon I've backed out the change to the Tests. Let me know if there is anything else you need.

**Comment by jrharmon** [comment]
Everything looks good to me.  I just noticed that PR #8 was never merged, but just approved.  I see you added an snk file, so there shouldn't be any need for that PR anymore, once this merges, right?  If so, I'll close that PR.

**Comment by jrharmon** [comment]
The AppVeyor build failed with the following error.

MockHttpServer.csproj(2,1): error MSB4041: The default XML namespace of the project must be the MSBuild XML namespace. If the project is authored in the MSBuild 2003 format, please add xmlns="http://schemas.microsoft.com/developer/msbuild/2003" to the <Project> element. If the project has been authored in the old 1.0 or 1.2 format, please convert it to MSBuild 2003 format.

**Comment by dansiegel** [comment]
As I mentioned in the PR notes, AppVeyor was failing, even when using the VS2017 image. I didn't have time to look further into what was causing it, there is nothing actually wrong with it, as it builds fine on locally and on VSTS.

**Comment by jrharmon** [comment]
Ok, thanks.  I saw your original comment, but saw a later commit that updated the AppVeyor build, and assumed that was what it was updating.  I'll take a look at it hopefully tomorrow, and try to get it working.

Thanks again for the work on this PR.

**Comment by dansiegel** [comment]
My pleasure, I look forward to seeing the updated package on NuGet.

**Comment by jrharmon** [comment]
So, I'm having trouble getting things working on my end.

I pulled down the latest to my Windows 7 machine, and had some build errors.  The first one was fixed by installing the latest 4.7 SDK from the VS installer, and the second one SHOULD have been fixed by installing the WIndows 10 SDK for UWP, but doesn't appear to be working.  I'm out of time tonight, so will try again hopefully soon on a Windows 10 machine, and see if that works better.

While I would love for this to be useable by as many targets as possible, I don't want to have too many requirements for building it.  Hopefully my Windows 10 machine has better luck.

**Comment by dansiegel** [comment]
@jrharmon there are a couple of requirements:

- SDK style projects require the use of VS2017
- netstandard2.0 requires having .NET 4.7+ installed
- UWP, I'm not sure is buildable from Windows 7. It will build on a Windows 10 machine, and should also build on Windows Server 2012R2+

**Comment by jrharmon** [comment]
I have all of those requirements, but the following folder didn't get created when adding in the UWP SDK.

C:\Program Files (x86)\Microsoft Visual Studio\2017\Professional\MSBuild\Microsoft\WindowsXaml

**Pull request merged**